flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/ContextClassLoadingSettingTest.java

```diff
@@ -150,21 +154,28 @@ void testAkkaRpcService_ScheduleRunnableSetsFlinkContextClassLoader()
     void testAkkaRpcService_ScheduleRunnableWithFixedRateSetsFlinkContextClassLoader() {
         final int numberOfScheduledRuns = 2;
         final List<ClassLoader> contextClassLoaders = new ArrayList<>(numberOfScheduledRuns);
-        akkaRpcService
-                .getScheduledExecutor()
-                .scheduleAtFixedRate(
-                        () -> {
-                            if (contextClassLoaders.size() < numberOfScheduledRuns) {
-                                contextClassLoaders.add(
-                                        Thread.currentThread().getContextClassLoader());
-                            } else {
-                                throw new RuntimeException("cancel task");
-                            }
-                        },
-                        0,
-                        1,
-                        TimeUnit.MILLISECONDS);
-
+        ScheduledFuture<?> future =
+                akkaRpcService
+                        .getScheduledExecutor()
+                        .scheduleAtFixedRate(
+                                () -> {
+                                    if (contextClassLoaders.size() < numberOfScheduledRuns) {
+                                        contextClassLoaders.add(
+                                                Thread.currentThread().getContextClassLoader());
+                                    } else {
+                                        throw new RuntimeException("cancel task");
+                                    }
+                                },
+                                0,
+                                1,
+                                TimeUnit.MILLISECONDS);
+        try {
+            future.get();
+            fail("The future should fail.");
```

Comment:
It could be more straight-forward to complete a future within `scheduleAtFixedRate` (before the throw the exception) that we wait for outside the executor.

Reply:
Hi @zentol , Thanks for your review, I updated.